CMakeLists.txt

```diff
@@ -114,6 +114,17 @@ file(RELATIVE_PATH RELATIVE_LLVM_TOOLS_BINARY_DIR "${CMAKE_INSTALL_PREFIX}"
      "${LLVM_TOOLS_BINARY_DIR}")
 file(APPEND "${ADDITIONAL_BIN_PATHS}" "${RELATIVE_LLVM_TOOLS_BINARY_DIR}\n")
 
+#
+# Component hash
+#
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/share/revng/component-hashes")
+add_custom_target(
+  compute_sha1 ALL
+  COMMAND git -C "${CMAKE_SOURCE_DIR}" rev-parse HEAD >
+          "${CMAKE_BINARY_DIR}/share/revng/component-hashes/revng")
+install(FILES "${CMAKE_BINARY_DIR}/share/revng/component-hashes/revng"
+        DESTINATION "${CMAKE_INSTALL_DIR}/share/revng/component-hashes/")
```

Comment:
1. Use a more descriptive target name. `ninja compute_sha1` is a bit obscure.
2. Are we already assume we are in a git repo in other parts of the codebase? If not, let's handle the situation in which we're not in a git repo.

Reply:
1. Is `compute_component_hash` ok?
2. What would you do? `dd if=/dev/random`?

Reply:
1. Yup
2. I'd put the project version according to CMake (`PROJECT_VERSION`)

Reply:
Both done